### DISCUSSION TITLE
Odoo 13 Addon: ValueError: Invalid field 'requested_date' on model 'sale.order'

### DISCUSSION THREAD

**Discussion by HTPit** [frePPLe/frepple#424] [Q&A] [answered]
Attempting Odoo 13 (no demo loaded) with frepple odoo 13 addon against Frepple 6.15.0. When testing the url
http://localhost:8069/frepple/xml?database=odoodb&language=en_US&company=YourCompany&mode=1

against Odoo 13, it returns the above error.
Searching against the now read-only Frepple discussion returned two threads which seemed related by inconclusive.
https://groups.google.com/g/frepple-users/search?q=requested_date

The actual outbound.py file states " The field requested_date is only available when sale_order_dates is installed."

Odoo 11 default sale module sale.py only has the following dates
https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py

    validity_date = fields.Date(string='Expiration Date', readonly=True, copy=False, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]},
        help="Manually set the expiration date of your quotation (offer), or it will set the date automatically based on the template if online quotation is installed.")
    is_expired = fields.Boolean(compute='_compute_is_expired', string="Is expired")
    create_date = fields.Datetime(string='Creation Date', readonly=True, index=True, help="Date on which sales order is created.")
    confirmation_date = fields.Datetime(string='Confirmation Date', readonly=True, index=True, help="Date on which the sales order is confirmed.", oldname="date_confirm", copy=False)
    

A search on the interweb returned this module for Odoo 11 with requested_date among others:
https://github.com/rythe77/odoo11_customized/blob/master/sale_requested_date/models/sale_order.py
  
    commitment_date = fields.Datetime(compute='_compute_commitment_date', string='Commitment Date', store=True,
                                      help="Date by which the products are sure to be delivered. This is "
                                           "a date that you can promise to the customer, based on the "
                                           "Product Lead Times.")
    requested_date = fields.Datetime('Delivery Date', readonly=True, states={'draft': [('readonly', False)],
                                     'sent': [('readonly', False)]}, copy=False,
                                     help="Date by which the customer has requested the items to be "
                                          "delivered.\n"
                                          "When this Order gets confirmed, the Delivery Order's "
                                          "expected date will be computed based on this date and the "
                                          "Company's Security Delay.\n"
                                          "Leave this field empty if you want the Delivery Order to be "
                                          "processed as soon as possible. In that case the expected "
                                          "date will be computed using the default method: based on "
                                          "the Product Lead Times and the Company's Security Delay.")
    effective_date = fields.Date(compute='_compute_picking_ids', string='Effective Date', store=True,
                                 help="Date on which the first Delivery Order was created.")

Assuming this is the referenced addon by Frepple, and give that there now new date fields available within Odoo13
https://github.com/odoo/odoo/blob/13.0/addons/sale/models/sale.py

    commitment_date = fields.Datetime('Delivery Date',
                                      states={'draft': [('readonly', False)], 'sent': [('readonly', False)]},
                                      copy=False, readonly=True,
                                      help="This is the delivery date promised to the customer. "
                                           "If set, the delivery order will be scheduled based on "
                                           "this date rather than product lead times.")
    expected_date = fields.Datetime("Expected Date", compute='_compute_expected_date', store=False,  # Note: can not be stored since depends on today()
        help="Delivery date you can promise to the customer, computed from the minimum lead time of the order lines.")

Can we just replace requested_date with commitment_date that is available within Odoo 13 by default? It seems closest to the what was originally intended.

**Comment by jdetaeye** [comment, marked as answer]
Your analysis is correct.   The field is from the sale_order_dates module. 
You can update the connector as you wish to send the sales order due date to frepple for planning.
